src/devices/Ad7193/Ad7193.cs

```diff
@@ -0,0 +1,842 @@
+                    _sb.Append(" | ");
+                    _sb.Append($"REFDET: {(registerValue & 0b0000_0000_0000_0000_0100_0000) >> 6}");
+                    _sb.Append(" | ");
+                    _sb.Append($"0: {(registerValue & 0b0000_0000_0000_0000_0010_0000) >> 5}");
+                    _sb.Append(" | ");
+                    _sb.Append($"BUF: {(registerValue & 0b0000_0000_0000_0000_0001_0000) >> 4}");
+                    _sb.Append(" | ");
+                    _sb.Append($"Unipolar: {(registerValue & 0b0000_0000_0000_0000_0000_1000) >> 3}");
+                    _sb.Append(" | ");
+                    _sb.Append($"Gain: {UInt32ToBinaryString((registerValue & 0b0000_0000_0000_0000_0000_0111), 3)}");
+
+                    result = _sb.ToString();
+                    break;
+                default:
+                    result = registerValue.ToString();
+                    break;
+            }
+
+            return result;
+        }
+
+        /// <summary>
+        /// Disposes the AD7193 object and closes the SPI device
+        /// </summary>
+        public void Dispose()
+        {
+            _spiDevice?.Dispose();
+            _spiDevice = null;
+        }
+
+        /// <summary>
+        /// Extracts a value from a register, where the value is located at the position the bitmask indicates
+        /// </summary>
+        /// <param name="reg">Register to extract the value from</param>
+        /// <param name="bitmask">Bitmask indicating the position of the value</param>
+        /// <returns>Value of the part of the register</returns>
+        private uint GetRegisterBits(Register reg, BitMask bitmask)
+        {
+            // clear all bit(s) except the bit(s) we would like to read
+            uint result = _registerCache[(byte)reg] & (uint)bitmask;
+
+            // remove the bits from the end of the register that are not part of the bitmask
+            if (bitmask > 0)
+            {
+                uint bits = (uint)bitmask;
+
+                while (bits % 2 == 0)
+                {
+                    result >>= 1;
+                    bits /= 2;
+                }
+            }
+
+            return result;
+        }
+
+        private uint SetRegisterBits(Register reg, BitMask bitmask, uint value)
+        {
+            // keep all bit values except the bit(s) we would like to modify
+            uint result = _registerCache[(byte)reg] & ~(uint)bitmask;
+
+            if (bitmask > 0)
+            {
+                uint bits = (uint)bitmask;
+
+                while (bits % 2 == 0)
+                {
+                    value <<= 1;
+                    bits /= 2;
+                }
+
+                value &= (uint)bitmask;
+                result |= value;
+            }
+
+            WriteRegisterValue(reg, result);
+
+            return result;
+        }
+
+        private uint ByteArrayToUInt32(byte[] buffer)
```

Comment:
please remove those helpers and use BinaryPrimitives instead which have utilities for all these